pkg/frontend/frontend.go

```diff
@@ -238,36 +238,43 @@ func (f *frontend) chiAuthenticatedRoutes(router chi.Router) {
 	r := router.With(f.authMiddleware.Authenticate)
 
 	r.Route("/subscriptions/{subscriptionId}", func(r chi.Router) {
-		r.Route("/resourcegroups/{resourceGroupName}/providers/{resourceProviderNamespace}/{resourceType}", func(r chi.Router) {
-			r.With(f.apiVersionMiddleware.ValidateAPIVersion).Get("/", f.getOpenShiftClusters)
-
-			r.Route("/{resourceName}", func(r chi.Router) {
-				r.With(f.apiVersionMiddleware.ValidateAPIVersion).Route("/", func(r chi.Router) {
-					// With API version check
-					if f.env.FeatureIsSet(env.FeatureEnableOCMEndpoints) {
-						r.Route("/{ocmResourceType}",
-							func(r chi.Router) {
-								r.Delete("/{ocmResourceName}", f.deleteClusterManagerConfiguration)
-								r.Get("/{ocmResourceName}", f.getClusterManagerConfiguration)
-								r.Patch("/{ocmResourceName}", f.putOrPatchClusterManagerConfiguration)
-								r.Put("/{ocmResourceName}", f.putOrPatchClusterManagerConfiguration)
-							},
-						)
-					}
-
-					r.Delete("/", f.deleteOpenShiftCluster)
-					r.Get("/", f.getOpenShiftCluster)
-					r.Patch("/", f.putOrPatchOpenShiftCluster)
-					r.Put("/", f.putOrPatchOpenShiftCluster)
-
-					r.Post("/listcredentials", f.postOpenShiftClusterCredentials)
-
-					r.Post("/listadmincredentials", f.postOpenShiftClusterKubeConfigCredentials)
+		r.Route("/resourcegroups/{resourceGroupName}/providers/{resourceProviderNamespace}", func(r chi.Router) {
```

Comment:
let's not mix the resource level route configuration with preflight api. instead, add one after this resourceType route configuration

Reply:
separated preflight route, and added after resourceType